proto/regen/ecocredit/v1/state.proto

```diff
@@ -83,6 +83,7 @@ message Project {
     index : {id : 2, fields : "class_key,id", unique : true}
     index : {id : 3, fields : "admin"}
     index : {id : 4, fields : "reference_id"}
+    index : {id : 5, fields : "admin,reference_id"}
```

Comment:
added this index to narrow down how many projects we have to sift through.

Reply:
that would definitely simplify things up a bit. however, what should we do for projects that don't specify a ref id? should we just set it to the project id?

Reply:
`WithAdmin(bridgeServiceAddr).WithReferenceId(req.ProjectRefId)` without the added index?

Reply:
I don't think we can make this unique given reference id has no unique constraints. We don't provide unique constraints because we want to avoid a situation where someone could potentially name squat a reference id that is not actually the reference id for the project.

Reply:
I don't think we can make this unique given reference id has no unique constraints. We added reference id without unique constraints because we want to avoid a situation where someone could potentially name squat a reference id.

Reply:
that's not possible unfortunately, the index is needed

Reply:
I don't think we can make this index unique given reference id has no unique constraints and reference id is only optional. We don't set unique constraints on reference id because we want to avoid a situation where someone could name squat a reference id and we don't require it because some projects may not have one.

Reply:
> could name squat a reference id
you wouldnt be able to name squat on a unique constraint on an admin+ref_id index key

>  because some projects may not have one

yeah, thats why i suggested we just set it to the generated project ID if they dont supply one, but not really sure how ideal that is

Reply:
> you wouldnt be able to name squat on a unique constraint on an admin+ref_id index key

True. Only an admin could name squat their own projects... which would be silly.

Reply:
> the generated project ID if they dont supply one

This could work. I'm guessing reference id would be associated with the `ClassProject` not the `Project` when we migrate away from projects being strictly defined within a credit class. Seems a bit strange to have the same generated project id stored twice for projects that do not provide a reference id. Curious if @clevinson has any input here.

Reply:
Let's not use the generated project ID for reference ID. It's intended to be an externally set identifier and optional, so I think its better if in state we reflect what the user's intent was (which is to not store a ref ID if they don't set it)

Reply:
ok ill revert the unique constraint changes. i think it might be safe to remove the constraint and error in the rare case we end up with two projects (i think this literally only happens if our keys are compromised and someone makes another project with the same ref id?). can flesh out further/other solutions on tomorrows call too 🙂